File: sandbox.client/front/ts/model/ClientToSave.ts
import {ClientAddress} from "./ClientAddress";
import {ClientPhones} from "./ClientPhones";
export class ClientToSave {
  public id: string;
  public name: string;
  public surname: string;
  public patronymic: string;
  public gender: string;
  public charmId: string;
  public dateOfBirth: string;
  public factAddress: ClientAddress;
  public regAddress: ClientAddress;
  public phones: ClientPhones;

  public assign(o: any): ClientToSave {
    o.phones.mobile = o.phones.mobile.filter(String);
    this.id = o.id;
    this.name = o.name;
    this.surname = o.surname;
    this.dateOfBirth = o.dateOfBirth;
    this.charmId = o.charmId;
    this.gender = o.gender;
    this.patronymic = o.patronymic;
    this.factAddress = o.factAddress;
    this.regAddress = o.regAddress;
    this.phones = o.phones;
    return this;
  }
}

File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/ClientToSave.java
package kz.greetgo.sandbox.controller.model;

import java.util.ArrayList;
import java.util.List;

public class ClientToSave {
  public String id;
  public String name;
  public String surname;
  public String patronymic;
  public String gender;
  public String charmId;
  public String dateOfBirth;
  public List<String> firstAddress = new ArrayList<>();
  public List<String> secondAddress = new ArrayList<>();
  public List<String> phones = new ArrayList<>();
}
